pkg/bootstrap/bootstrap.go

```diff
@@ -258,20 +260,40 @@ func (c *BootstrapClient) applyManifests(
 			}
 		}
 
-		if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		applyFn := func() error {
 			_, err = c.Client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
 				Apply(ctx, obj.GetName(), &obj, getApplyOptions(options))
 			return err
-		}); err != nil && (!options.DryRun || !apierrors.IsNotFound(err)) {
-			// we can recover from dry-run errors appearing because an old job has not been deleted
-			var statusErr *apierrors.StatusError
-			if options.DryRun && options.Force && obj.GetKind() == constants.Job && errors.As(err, &statusErr) {
-				if statusErr.ErrStatus.Status == "Failure" && statusErr.ErrStatus.Reason == "Invalid" {
-					statusMessage("Ignoring Job immutable error in dry-run", true, false)
-					return nil
+		}
+
+		if obj.GetKind() == "PackageRepository" {
+			// when updating an existing installation, new certs are generated and might not be ready yet.
+			// in that case, there will be an error from the kubernetes api with status 500, that has a message like this:
+			// failed calling webhook "vpackagerepository.kb.io": failed to call webhook: Post "https://.../validate-...":
+			// tls: failed to verify certificate: x509: certificate signed by unknown authority
+			// however, there can apparently also be other errors like "resource not found" wrapped in the internal error,
+			// so instead of explicitly looking for the cert error message, we simply retry in all internal error cases
+			if err = retry.OnError(wait.Backoff{
+				Duration: 2 * time.Second,
+				Factor:   3,
+				Jitter:   0.1,
+				Steps:    5,
+			}, apierrors.IsInternalError, applyFn); err != nil {
```

Comment:
Is there a reason for not using `retry.DefaultRetry`? Because if that would also be possible, we could simplify this to (something like):
```go
retry.OnError(
  retry.DefaultRetry, 
  func(err error) bool { return apierrors.IsInternalError(err) || apierrors.IsConflict(err) },
  applyFn,
)
```

… and then possibly inline applyFn

Reply:
We could also create a `predicate` package with the following generic function to make this kind of thing more ergonomic:
```
func All[T any](fns ...func(T) bool) func(T) bool {
	return func(arg T) bool {
		for _, fn := range fns {
			if fn(arg) {
				return true
			}
		}
		return false
	}
}
```

Reply:
OK, but I'm still not a fan of having different retry logic based on the resource kind. E.g. what if there would be a conflict for the `PackageRepository`. I think we could also nest the retry calls like this:

```go
err = retry.OnError(InternalErrorBackoff, apierrors.IsInternalError, 
  retry.RetryOnConflict(retry.DefaultRetry, func() error {
    // ...
  })
)
```

What do you think?